controllers/posts_controller.js

```diff
@@ -69,59 +63,36 @@ module.exports.search = (req, res) => {
     }
 
     if (limit)
-        req.query.limit = parseInt(limit);
+        req.query.limit = Number(limit);
 
     if (tags)
         req.query.tags = tags instanceof Array ? tags : [tags];
 
     req.query.sort = sort ? JSON.parse(sort) : {};
 
-    const searchObservable = post.search(req.query);
-    searchObservable.subscribe(
-        posts => res.status(200).json({
+    try {
```

Comment:
not a fan of the `try{} catch {}` design, but that's personal taste. I'd have the model methods return an error so you can do logic with that (that is the idiom in Go):

``` javascript
const {posts, err} = await post.search(req.query);
if (err) {
  console.error(err);
  res.status(500).json({

  });
  return;
}

// business logic without error here
```

Reply:
Hmmm, I'm been enjoying using try {} catch() {}

Reply:
Plus I think try catch is how you deal with promise rejections with async/await

Reply:
Yeah cause what happens if post.search throws an exception

Reply:
well rather than throwing an exception, `post.search` returns the error as a property, as you can see in the example. You can deconstruct the response easily that way:

`const {posts, err} = await post.search(req.query);`

Reply:
Then won't we have to handle exceptions in post.search?